src/core/core.controller.js

```diff
@@ -25,37 +43,37 @@ function mergeScaleConfig(config, options) {
 	const scales = {};
 
 	// First figure out first scale id's per axis.
-	// Note: for now, axis is determined from first letter of scale id!
 	Object.keys(configScales).forEach(id => {
-		const axis = id[0];
+		const scaleConf = configScales[id];
+		const axis = scaleConf.axis || id[0];
 		firstIDs[axis] = firstIDs[axis] || id;
-		scales[id] = helpers.mergeIf({}, [configScales[id], chartDefaults.scales[axis]]);
+		scales[id] = helpers.mergeIf({axis}, [scaleConf, chartDefaults.scales[axis]]);
 	});
 
 	// Backward compatibility
 	if (options.scale) {
-		scales[options.scale.id || 'r'] = helpers.mergeIf({}, [options.scale, chartDefaults.scales.r]);
+		scales[options.scale.id || 'r'] = helpers.mergeIf({axis: 'r'}, [options.scale, chartDefaults.scales.r]);
 		firstIDs.r = firstIDs.r || options.scale.id || 'r';
 	}
 
 	// Then merge dataset defaults to scale configs
 	config.data.datasets.forEach(dataset => {
-		const datasetDefaults = defaults[dataset.type || config.type] || {scales: {}};
+		const type = dataset.type || config.type;
+		const baseAxis = dataset.baseAxis || getBaseAxis(type, options);
+		const datasetDefaults = defaults[type] || {};
 		const defaultScaleOptions = datasetDefaults.scales || {};
 		Object.keys(defaultScaleOptions).forEach(defaultID => {
-			const id = dataset[defaultID + 'AxisID'] || firstIDs[defaultID] || defaultID;
+			const axis = getAxisFromDefaultScaleID(defaultID, baseAxis);
+			const id = dataset[axis + 'AxisID'] || firstIDs[axis] || axis;
 			scales[id] = scales[id] || {};
-			helpers.mergeIf(scales[id], [
-				configScales[id],
-				defaultScaleOptions[defaultID]
-			]);
+			helpers.mergeIf(scales[id], [{axis}, configScales[id], defaultScaleOptions[defaultID]]);
 		});
 	});
 
 	// apply scale defaults, if not overridden by dataset defaults
 	Object.keys(scales).forEach(key => {
 		const scale = scales[key];
-		helpers.mergeIf(scale, scaleService.getScaleDefaults(scale.type));
+		helpers.mergeIf(scale, [chartDefaults.scales[scale.axis], scaleService.getScaleDefaults(scale.type)]);
```

Comment:
In the case of the bar chart, the defaults are `i` and `v`, so wouldn't this not pick up anything from chartDefaults?

Reply:
In every step above, the `axis` is merged to scale. So at this point its going to be one of `'x', 'y' or 'r'`
So no, the defaults from `i` / `v` are not going to be merged here. But if there is a dataset, then those were merged already in the loop above. And if there is a scale without dataset, then the first loop gets the defaults (using scale.axis or scale.id[0]).

Reply:
Actually not so sure if this is needed at all now (the chartDefaults, will try)

Reply:
Removed